RELEASE_MANAGEMENT.md

```diff
@@ -0,0 +1,107 @@
+# Release Management
+
+Major and minor releases for Gateway API are managed by a "Release Manager".
+The responsibilities of the release manager include:
+
+* Creating and managing a GitHub Milestone for the release.
+* Creating and managing a GitHub Project board for the release.
+* Creating and managing a GitHub Discussion Boards announcement for the release.
```

Comment:
This isn't just for announcement of a release, we also use discussion boards to handle release scoping for each release channel.

Reply:
```suggestion
* Creating and managing a GitHub Discussion Boards announcement for the release.
* Creating and managing discussions to handle release scoping for each release channel.
```